el_reward_engine.go

```diff
@@ -0,0 +1,222 @@
+package relayproxy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/bloXroute-Labs/relayproxy/fluentstats"
+	"github.com/rs/zerolog"
+)
+
+const (
+	proposerHeaderDeliveredURI = "/relay/v1/data/bidtraces/proposer_header_delivered?slot="
+)
+
+type RewardEngine struct {
+	logger               zerolog.Logger
+	httpClient           *http.Client
+	slotStartRecordCh    chan SlotStatsRecord
+	relayUrlsWithApiKeys map[string]string
+	nodeID               string
+	fluentd              fluentstats.Stats
+}
+
+type BidTrace struct {
```

Comment:
Can we call this `DataHeaders` or something similar to match https://github.com/bloXroute-Labs/mev-boost-relay/blob/develop/server/bx_types.go#L29